backend/index.js

```diff
@@ -44,71 +56,133 @@ io.on("connection", (socket) => {
     socket.to(data.room).emit("receive_message", data);
   });
 
-	socket.on("send_image", (data) => {
-		console.log("Image received from", data.author, data.room);
+  socket.on("send_image", (data) => {
+    console.log("Image received from", data.author, data.room);
     socket.to(data.room).emit("receive_image", data);
     console.log("Image sent to", data.room);
-	});
+  });
 
   socket.on("disconnect", () => {
     console.log("A user has left the platform");
   });
 });
 
+// Custom Auth Middleware
+const authorization = (req, res, next) => {
+  const token = req.cookies.access_token;
+  if (!token) {
+    return res
+      .status(403)
+      .json({ message: "Proper authentication not supplied" });
+  }
+  try {
+    const data = jwt.verify(token, JWT_SECRET);
+    req.room = data.room;
+    req.room = data.role;
+    return next();
+  } catch {
+    return res.status(403);
+  }
+}
+
 // REST Endpoints
 app.get('/', (req, res) => {
-  res.json({
-    rooms: roomStore
-  })
+  return res
+    .status(200)
+    .json({
+      rooms: roomStore
+    });
 });
 
 app.get('/login', async (req, res) => {
```

Comment:
- [ ] We should be logging when admins log in for security auditing. **WE SHOULD NOT** Log the password used, just valid and invalid login attempts.

Reply:
I added line #113 to address this
```js
console.log(username, "logged in", token);
```